icu4c/source/i18n/number_longnames.cpp

```diff
@@ -39,19 +39,36 @@ constexpr int32_t DNAM_INDEX = StandardPlural::Form::COUNT;
  * `per` forms.
  */
 constexpr int32_t PER_INDEX = StandardPlural::Form::COUNT + 1;
+/**
+ * Gender of the word, in languages with grammatical gender.
+ */
+constexpr int32_t GENDER_INDEX = StandardPlural::Form::COUNT + 2;
 // Number of keys in the array populated by PluralTableSink.
-constexpr int32_t ARRAY_LENGTH = StandardPlural::Form::COUNT + 2;
+constexpr int32_t ARRAY_LENGTH = StandardPlural::Form::COUNT + 3;
 
 static int32_t getIndex(const char* pluralKeyword, UErrorCode& status) {
```

Comment:
Following examples from standardplural.cpp
We do have only three things to check here, whereas for both examples in standardplural.cpp it's six. Potentially not worth having a switch if we're only checking for 3 values? Shall I revert the commit in which I did this?

Reply:
:man_shrugging: